### PR TITLE
Issue #6397: upgraded picocli to 3.9.5

### PR DESCRIPTION
**Pull request by rnveach**
Issue #6397

@romani Do we have a suppression somewhere that prevented this library from being upgraded until the issue was resolved?

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 1]
```diff
@@ -292,7 +292,8 @@ public void testNonExistentOutputFormat() throws Exception {
         exit.checkAssertionAfterwards(() -> {
             assertEquals("Unexpected output log", "", systemOut.getLog());
             assertEquals("Unexpected system error log",
-                    "Invalid value for option '-f': expected one of [xml, plain] but was 'xmlp'"
+                    "Invalid value for option '-f': expected one of [XML, PLAIN]"
+                        + " (case-insensitive) but was 'xmlp'"
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: This was the reason for the failure.

`checkAssertionAfterwards` must execute before `expectSystemExitWithStatus`. When `checkAssertionAfterwards` fails it throws an AssertionException and this probably triggers another `System.exit` which then `expectSystemExitWithStatus` catches and tries to match against.

Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: According to `ExpectedSystemExit` it checks system exit before the output assertions.

Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: @romani Somehow powermock is again interfering when the junit results in this case. Removing the powermock annotations, I get the following junit failure:
````
[INFO] Running com.puppycrawl.tools.checkstyle.MainTest
[ERROR] Tests run: 73, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 5.611 s <<< FAILURE! - in com.puppycrawl.tools.checkstyle.MainTest
[ERROR] testJacocoWorkaround(com.puppycrawl.tools.checkstyle.MainTest)  Time elapsed: 0.035 s  <<< FAILURE!
java.lang.AssertionError: Unexpected call of System.exit(-1).

[ERROR] testNonExistentOutputFormat(com.puppycrawl.tools.checkstyle.MainTest)  Time elapsed: 0.25 s  <<< FAILURE!
org.junit.ComparisonFailure: 
Unexpected system error log expected:<...': expected one of [[xml, plain]] but was 'xmlp'
Usa...> but was:<...': expected one of [[XML, PLAIN] (case-insensitive)] but was 'xmlp'
Usa...>
	at com.puppycrawl.tools.checkstyle.MainTest.lambda$testNonExistentOutputFormat$7(MainTest.java:294)
````
I can only assume powermock is trying to take this exception as it's own and mangle it in the process.

**Comment by romani** [comment]
https://github.com/checkstyle/checkstyle/blob/master/config/version-number-rules.xml#L31

**Comment by rnveach** [comment]
> https://github.com/checkstyle/checkstyle/blob/master/config/version-number-rules.xml#L31

Done.

**Pull request merged**